RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -38,34 +69,52 @@ public RestClient(ConnectionDetails connectionDetails) {
             });
     }
 
+    @WithSpan
     public Mono<Response> getAsync(String path) {
-        return client.get()
+        return client
+            .doOnRequest(sizeMetricsHandler("GET", path))
+            .get()
             .uri("/" + path)
             .responseSingle((response, bytes) -> bytes.asString()
-                .map(body -> new Response(response.status().code(), body, response.status().reasonPhrase())));
+            .map(body -> new Response(response.status().code(), body, response.status().reasonPhrase())))
+            .doOnError(Span.current()::recordException);
```

Comment:
I never actually spun up monitoring on this so this was a debug todo

_Deferring since we aren't going to move forward with automatic instrumentation_